azure/scope/machinepool.go

```diff
@@ -211,8 +211,11 @@ func (m MachinePoolScope) MaxSurge() (int, error) {
 // updateReplicasAndProviderIDs ties the Azure VMSS instance data and the Node status data together to build and update
 // the AzureMachinePool replica count and providerIDList.
 func (m *MachinePoolScope) updateReplicasAndProviderIDs(ctx context.Context) error {
-	ctx, span := tele.Tracer().Start(ctx, "scope.MachinePoolScope.UpdateInstanceStatuses")
-	defer span.End()
+	ctx, _, done := tele.StartSpanWithLogger(
+		ctx,
```

Comment:
Nitpick: I think this reads more easily as a one-liner, which it is in most of the changes here.

Reply:
@mboersma done!

Reply:
doesn't this apply to the other files as well?

Reply:
It does--there are several more below that could be one-liners, but I don't want to block this PR on trivia.